src/index.js

```diff
@@ -432,6 +432,36 @@ class MiniCssExtractPlugin {
           return source;
         }
       );
+
+      compilation.hooks.optimizeChunks.tap(pluginName, (chunks) => {
+        for (const chunk of chunks) {
+          chunk.modulesIterable.forEach((module) => {
+            // eslint-disable-next-line no-param-reassign
+            module.dependencies = module.dependencies.filter((dep) => {
+              const hasCssDep =
+                dep.module &&
+                dep.module.dependencies.find(
+                  (d) => d.module && d.module.type === MODULE_TYPE
+                );
+
+              if (hasCssDep) {
+                dep.disconnect();
+                return false;
+              }
+
+              return true;
+            });
+
+            if (
+              module.type === MODULE_TYPE &&
+              module.issuer &&
```

Comment:
A module can be referenced by multiple modules, but there is only one issuer.

Avoid using issuer.